File: 2020-Python/Day1.py
import sys

# Array to store entries from file
entries = []

# Input file
infile = open("Day1_input.txt", 'r')

# Load file to entries
for line in infile:
    entries.append(int(line))
infile.close()

# Find Part 1 Answer
for x in range(len(entries)):
    for y in range(x+1, len(entries)):
        if entries[x] + entries[y] == 2020:
            print("Part 1 Answer:", entries[x]*entries[y])

# Find Part 2 Answer
for x in range(len(entries)):
    for y in range(x+1, len(entries)):
        for z in range(y+1, len(entries)):
            if entries[x] + entries[y] + entries[z]  == 2020:
                print("Part 2 Answer:", entries[x]*entries[y]*entries[z])

File: 2022-C++/src/day1.cpp
#include <string>
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>
#include <numeric>

using namespace std;

int main(int, char**) {

    fstream ifs;
    string line_str;
    vector<int> calories;

    ifs.open("input/day1_input.txt", ios::in);

    int calories_for_elf = 0;

    while (!ifs.eof()) {
        getline(ifs, line_str);

        if (line_str.empty()) {
            calories.push_back(calories_for_elf);
            calories_for_elf = 0;
        }
        else {
            calories_for_elf += stoi(line_str);
        }
    }

    ifs.close();

    sort(calories.begin(), calories.end());
    cout << "Part 1: " << calories.back() << endl;

    auto sum = accumulate(calories.end() -3, calories.end(), 0);    
    cout << "Part 2: " << sum << endl;

    return 0;
}